src/votekit/elections/election_types/scores/rating.py

```diff
@@ -0,0 +1,197 @@
+from ....models import Election
+from ....pref_profile import PreferenceProfile
+from ...election_state import ElectionState
+from ....utils import (
+    score_profile_from_ballot_scores,
+    elect_cands_from_set_ranking,
+    remove_cand,
+)
+from typing import Optional, Union
+from fractions import Fraction
+
+
+class GeneralRating(Election):
+    """
+    General rating election. To fill :math:`m` seats, voters score each candidate from
+    :math:`0-L`, where :math:`L` is some user-specified limit.  There is also a total budget of
+    :math:`k` points per voter. The :math:`m` winners are those with the highest total score.
+
+    Args:
+        profile (PreferenceProfile): Profile to conduct election on.
+        m (int, optional): Number of seats to elect. Defaults to 1.
+        L (Union[float, Fraction], optional): Rating per candidate limit. Defaults to 1.
+        k (Union[float, Fraction], optional): Budget per ballot limit. Defaults to None, in which
+            case voters can score each candidate independently.
+        tiebreak (str, optional): Tiebreak method to use. Options are None and 'random'.
+            Defaults to None, in which case a tie raises a ValueError.
+
+    """
+
+    def __init__(
+        self,
+        profile: PreferenceProfile,
+        m: int = 1,
+        L: Union[float, Fraction] = 1,
+        k: Optional[Union[float, Fraction]] = None,
+        tiebreak: Optional[str] = None,
+    ):
+        self.m = m
+        if L <= 0:
+            raise ValueError("L must be positive.")
+        self.L = L
```

Comment:
Shouldn't we also raise an error if L > k? The `_validate_profile` method will raise an error if you go above the total score budget, but it doesn't make sense to allow scores higher than the total budget